### ISSUE TITLE
Error: invalid kube version '1.25+': Invalid Semantic Version

### ISSUE THREAD

**Issue by bersanf** [gardener/garden-setup#1197]
**What happened**:
I am trying to install gardener on Amazon EKS (with k8s 1.25) cluster following the [installation guide](https://github.com/gardener/garden-setup).

When i execute the command `sow deploy -A` i get the following error:
_Error: invalid kube version '1.25+': Invalid Semantic Version_

The sow command is executing the following step:
_Running helm template for garden-setup-dns-controller-manager ...
helm template garden-setup-dns-controller-manager --kubeconfig "/path/to/landscape/gen/dns-controller/helm/kubeconfig" --values "/path/to/landscape/gen/dns-controller/helm/values.json" --namespace kube-system --kube-version=1.25+ "/path/to/landscape/gen/dns-controller/git/repo/charts/external-dns-management"_

**What you expected to happen**:
The installation should complete without errors

**How to reproduce it (as minimally and precisely as possible)**:
To reproduce you can provision an Amazon Cloud9 environment with ubuntu 22.04 and follow the steps.

**Anything else we need to know?**:

**Environment**:

- Version of garden-setup (release or commit): https://github.com/gardener/garden-setup/commit/9403d6815080c93fe87cf333573875f20660bfce
- Versions of components (only needed if you overwrote the defaults):
Kubernetes 1.25
kubectl client: 1.28.1
helm: v3.12.3+g3a31588
 
- Where does the underlying base cluster come from and which operating system does it use?
bottlerocket for kubernetes nodes
- Which cloud provider is configured for the setup? AWS
- Other:

**Comment by Diaphteiros** [comment]
Hm, interesting. Let me explain what happens there:

`helm` can be passed a `--kube-version` argument when templating manifests. This is relevant in a very few cases where the k8s api changed and the way helm renders manifests has to adapt to the cluster version.

In order to get the version, there is a dummy component which doesn't do anything except for running `kubectl version -o yaml` and parsing the result. Usually, it should look something like this:
```yaml
kubectl version -o yaml
clientVersion:
  buildDate: "2023-06-14T09:47:38Z"
  compiler: gc
  gitCommit: 25b4e43193bcda6c7328a6d147b1fb73a33f1598
  gitTreeState: clean
  gitVersion: v1.27.3
  goVersion: go1.20.5
  major: "1"
  minor: "27"
  platform: darwin/arm64
kustomizeVersion: v5.0.1
serverVersion:
  buildDate: "2023-07-19T12:17:23Z"
  compiler: gc
  gitCommit: ba490f01df1945d0567348b271c79a2aece7f623
  gitTreeState: clean
  gitVersion: v1.25.12
  goVersion: go1.20.6
  major: "1"
  minor: "25"
  platform: linux/amd64
```

I assume that your cluster returns `25+` as `minor` version for some reason. Could you run `kubectl version -o yaml` against your cluster to verify this?

I think I will implement a possibility to set this value in the acre.yaml file instead of fetching it from the cluster.

In the meantime, if you want to continue, here is a quick and dirty workaround:
Manually edit `export/k8sversion/export.yaml` and remove the trailing `+` from all versions in there. Note that this file will be overwritten whenever the `k8sversion` component is deployed.

**Comment by bersanf** [comment]
hello Diaptheiros
thank you for your feedback.
You are totally right, the server returns 25+ as minor version.

Here the output of command `kubectl version -o yaml`:

_clientVersion:
  buildDate: "2023-08-15T16:08:20Z"
  compiler: gc
  gitCommit: a774c94033623662da4c383a69c1ce10c0bdf9fa
  gitTreeState: clean
  gitVersion: v1.25.12-eks-8ccc7ba
  goVersion: go1.20.6
  major: "1"
  minor: 25+
  platform: linux/amd64
kustomizeVersion: v4.5.7
serverVersion:
  buildDate: "2023-07-28T16:52:47Z"
  compiler: gc
  gitCommit: af930c12e26ef9d1e8fac7e3532ff4bcc1b2b509
  gitTreeState: clean
  gitVersion: v1.24.16-eks-2d98532
  goVersion: go1.20.6
  major: "1"
  minor: 24+
  platform: linux/amd64_

**Comment by bersanf** [comment]
Editing the  `export/k8sversion/export.yaml` did not solve, because as you said at every deployment the export.yaml file gets overridden.

**Comment by bersanf** [comment]
I was able to continue the installation by installing single components without dependencies.

**Comment by Diaphteiros** [comment]
Fixed with #1203
